### PR TITLE
bpo-35078:Allow customization of CSS class name of a month in calendar module

### PR DESCRIPTION
**Pull request by srinivasreddy**
https://bugs.python.org/issue35078

### CODE REVIEW

File: Lib/calendar.py [hunk 1]
```diff
@@ -609,7 +609,7 @@ def formatmonthname(self, theyear, themonth, withyear=True):
             s = month_name[themonth]
             if withyear:
                 s = '%s %s' % (s, theyear)
-            return '<tr><th colspan="7" class="month">%s</th></tr>' % s
+            return '<tr><th colspan="7" class="%s">%s</th></tr>' % (self.cssclass_month_head, s)
```
Comment (tirkarthi) [inline Lib/calendar.py]: I don't know why the logic in base class is repeated and maybe I am missing something. How about the below ?

```python
    def formatmonthname(self, theyear, themonth, withyear=True):
        with different_locale(self.locale):
            return super().formatmonthname(theyear, themonth, withyear)
```

**Comment by tirkarthi** [comment]
Tests could be added for the new change at https://github.com/python/cpython/blob/4e3a53bceefe4803c08a025523d8658862cb31c0/Lib/test/test_calendar.py#L561

```python
self.assertIn('class="month"', local_month)
cal.cssclass_month_head = "text-center month"
local_month = cal.formatmonthname(2010, 10)
self.assertIn('class="text-center month"', local_month)
```

A NEWS entry might help with the change.

Thanks for the PR :)

**Review by tirkarthi — APPROVED** [review]
I would suggest adding the actual change done to the NEWS entry in addition to the current text like 

```markdown
The month's head CSS class in :class:`calendar.LocaleHTMLCalendar`
is now customizable with attribute ``cssclass_month_head``.
```

It's more of a personal opinion and would leave it to the reviewer to consider the wording. Other than that LGTM. Thanks for your contribution :)

**Comment by doerwalter** [comment]
The code looks good to me. There's no additional test for `LocaleTextCalendar`, but that's ok, as there should be no behaviour changes from the patch.

However there's no test for the changed method `LocaleHTMLCalendar.formatweekday`.

Furthermore the NEWS entry is misleading (and as Serhiy noted on bpo: "We don't usually do pure refactoring changes"). I'd use something like: "Refactor `formatweekday`, `formatmonthname` methods in `LocaleHTMLCalendar` and `LocaleTextCalendar` classes in `calendar` module to call the base class methods. This enables customizable CSS classes for `LocaleHTMLCalendar`.
Patch by Srinivas Reddy Thatiparthy."

If you could add a test for `LocaleHTMLCalendar.formatweekday` and update the NEWS entry, IMHO the patch is good to go.

**Comment by srinivasreddy** [comment]
Sure @doerwalter

**Comment by doerwalter** [comment]
There's still a test missing for the changed `formatweekday` method, i.e. something like:

```diff
-    def test_locale_html_calendar_custom_css_class(self):
+    def test_locale_html_calendar_custom_css_class_month_name(self):
         try:
             cal = calendar.LocaleHTMLCalendar(locale='')
             local_month = cal.formatmonthname(2010, 10, 10)
@@ -576,6 +576,19 @@ class CalendarTestCase(unittest.TestCase):
         local_month = cal.formatmonthname(2010, 10, 10)
         self.assertIn('class="text-center month"', local_month)

+    def test_locale_html_calendar_custom_css_class_weekday(self):
+        try:
+            cal = calendar.LocaleHTMLCalendar(locale='')
+            local_weekday = cal.formatweekday(6)
+        except locale.Error:
+            # cannot set the system default locale -- skip rest of test
+            raise unittest.SkipTest('cannot set the system default locale')
+        self.assertIn('class="sun"', local_weekday)
+        cal.cssclasses_weekday_head = [ "mon2", "tue2", "wed2", "thu2",
+          "fri2", "sat2", "sun2"]
+        local_weekday = cal.formatweekday(6)
+        self.assertIn('class="sun2"', local_weekday)
+
```

 Otherwise the patch looks good.

**Review by doerwalter — COMMENTED** [review]
See my last comment.

**Comment by csabella** [comment]
@srinivasreddy, please the last code review request.  Thank you!

**Comment by srinivasreddy** [comment]
@csabella ✅ Thanks

**Pull request merged**